gems/pending/util/mount/miq_glusterfs_session.rb

```diff
@@ -0,0 +1,35 @@
+require 'util/mount/miq_generic_mount_session'
+
+class MiqGlusterfsSession < MiqGenericMountSession
+  PORTS = [2049, 111]
+
+  def initialize(log_settings)
+    super(log_settings.merge(:ports => PORTS))
+  end
+
+  def connect
+    scheme, userinfo, @host, port, registry, @mount_path, opaque, query, fragment = URI.split(URI.encode(@settings[:uri]))
+    super
+  end
+
+  def mount_share
+    super
+
+    log_header = "MIQ(#{self.class.name}-mount_share)"
+    logger.info("#{log_header} Connecting to host: [#{@host}], share: [#{@mount_path}] using mount point: [#{@mnt_point}]...")
+
+    mount = "mount"
+    mount << " -r" if settings_read_only?
+
+    # Quote the host:exported directory since the directory can have spaces in it
+    case Sys::Platform::IMPL
+    when :macosx
+      runcmd("sudo #{mount} -t glusterfs -o resvport '#{@host}:#{@mount_path}' #{@mnt_point}")
+    when :linux
```

Comment:
I would keep this in for now. If it is an issue, it will only be encountered in a development environment.